src/requests.ts

```diff
@@ -0,0 +1,182 @@
+    Int8Array, Uint8Array, Int16Array, Uint16Array,
+    Int32Array, Uint32Array, Float32Array, Float64Array
+  ];
+  isArrayBufferView = function (object): boolean {
+    for (let i = 0; i < 8; i++) {
+      if (object instanceof arrayBufferViews[i]) {
+        return true;
+      }
+    }
+    return false;
+  };
+}
+
+
+export interface RequestOptions {
+  body?: XMLHttpRequestBodyInit;
+  headers?: HeadersInit;
+  responseType?: XMLHttpRequestResponseType;
+}
+
+export async function requestWithTimeout(method: string, url: string, options: RequestOptions): Promise<XMLHttpRequest> {
+  if (typeof fetch === 'function') {
+    return _fetchRequestWithTimeout(method, url, options);
+  } else if (typeof XMLHttpRequest === 'function') {
+    return _xhrRequestWithTimeout(method, url, options);
+  } else {
+    return Promise.reject('[Requests] You need to add a polyfill for fetch or XMLHttpRequest');
+  }
+}
+
+async function _fetchRequestWithTimeout(method: string, url: string, options: RequestOptions): Promise<XMLHttpRequest> {
+  const abortController = typeof AbortController === 'function' ?
+    new AbortController() :
+    null;
+  let timeoutId;
+
+  const timeoutPromise: Promise<never> = new Promise((_resolve, reject) => {
+    timeoutId = setTimeout(() => {
+      if (abortController) {
+        abortController.abort();
+      }
+      reject('timeout');
+    }, config.requestTimeout);
+  });
+
+  let syntheticXhr;
+  const responseHeaders = {};
+
+  const networkPromise: Promise<XMLHttpRequest> = fetch(url, {
+    method: method,
+    headers: options.headers,
+    body: options.body,
+    signal: abortController ? abortController.signal : undefined
+  }).then((response) => {
+    log('[requests fetch]', response);
+
+    response.headers.forEach((value: string, headerName: string) => {
+      responseHeaders[headerName.toUpperCase()] = value;
+    });
+
+    syntheticXhr = {
+      readyState: 4,
+      status: response.status,
+      statusText: response.statusText,
+      response: undefined,
+      getResponseHeader: (headerName: string): string => {
+        return responseHeaders[headerName.toUpperCase()] || null;
+      },
+      // responseText: 'foo',
+      responseType: options.responseType,
+      responseURL: url,
+    };
+    switch (options.responseType) {
+      case 'arraybuffer':
+        return response.arrayBuffer();
+      case 'blob':
+        return response.blob();
+      case 'json':
+        return response.json();
+      case undefined:
+      case '':
+      case 'text':
+        return response.text();
+      default:   // document
+        throw new Error("responseType 'document' is not currently supported using fetch");
+    }
+  }).then((processedBody) => {
+    syntheticXhr.response = processedBody;
+    if (!options.responseType || options.responseType === 'text') {
+      syntheticXhr.responseText = processedBody;
+    }
+    return syntheticXhr;
+  }).finally(() => {
+    clearTimeout(timeoutId);
```

Comment:
This is the only significant difference between the functions here, and the old ones in wireclient.ts. Node isn't able to clean up obsolete timeouts.